mlem/core/dataset_type.py

```diff
@@ -159,7 +218,7 @@ class SizedTypedListType(ListTypeWithSpec):
     """
 
     dtype: DatasetType
-    size: Optional[int]
+    size: int
```

Comment:
`size = None` means dataset can have any length (but still all the elements should be of the same type)